python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -109,8 +129,19 @@ def _kernel(retval, {input_columns}, {input_offsets}, size):
     i = cuda.grid(1)
     ret_data_arr, ret_mask_arr = retval
     if i < size:
+        # Create a structured array with the desired fields
+        rows = cuda.local.array(1, dtype=row_type)
+
+        # one element of that array
+        row = rows[0]
+
 {masked_input_initializers}
-        ret = {user_udf_call}
+{row_initializers}
+
+        # pass the abstract row into the udf
```

Comment:
In what way is the row "abstract"? Isn't it more like:
```suggestion
        # pass the assembled row into the udf
```
?